zerver/lib/outgoing_webhook.py

```diff
@@ -36,32 +37,48 @@ def __init__(self, base_url, token, user_profile, service_name):
     # The return value should be a tuple (rest_operation, request_data), where:
     # rest_operation is a dictionary containing atleast the following keys: method, relative_url_path and
     # request_kwargs. It provides rest operation related info.
-    # request_data is a dictionary whose format can vary depending on the type of webhook service.
+    # request_data is the data to be sent to third party service.
+    # It should be in the appropriate form. Ex: certain services require the post data to be in
+    # json form. Hence request_data returned here should be a json.
     def process_event(self, event):
-        # type: (Dict[str, Any]) -> Tuple[Dict[str ,Any], Dict[str, Any]]
+        # type: (Dict[str, Any]) -> Tuple[Dict[str, Any], Any]
         raise NotImplementedError()
 
-    # Given a successful response to the outgoing webhook REST operation, returns the message
-    # that should be sent back to the user.
+    # Given a successful response to the outgoing webhook REST operation, determines if response is required
+    # to be sent to user and in such case, returns the message that should be sent back.
     #
     # The response will be the response object obtained from REST operation.
     # The event will be the same as the input to process_command.
-    # The returned message will be a dictionary which should have "response_message" as key and response message to
-    # be sent to user as value.
+    # It returns the message to be sent back to user. If None is returned back, no response is sent back.
     def process_success(self, response, event):
-        # type: (Response, Dict[Text, Any]) -> Dict[str, Any]
+        # type: (Response, Dict[Text, Any]) -> Optional[str]
         raise NotImplementedError()
 
-    # Given a failed outgoing webhook REST operation, returns the message that should be sent back to the user.
+    # Given a failed outgoing webhook REST operation, determines if response is required
+    # to be sent to user and in such case, returns the message that should be sent back.
     #
     # The response will be the response object obtained from REST operation.
     # The event will be the same as the input to process_command.
-    # The returned message will be a dictionary which should have "response_message" as key and response message to
-    # be sent to user as value.
+    # It returns the message to be sent back to user. If None is returned back, no response is sent back.
     def process_failure(self, response, event):
-        # type: (Response, Dict[Text, Any]) -> Dict[str, Any]
+        # type: (Response, Dict[Text, Any]) -> Optional[str]
         raise NotImplementedError()
 
+    # Make a message more readable by refactoring @-mentions in a string. EX: it replaces @**vabs22** to
+    # @vabs22 in the message.
+    # The interface classes can also overwrite this function to create their own implementation.
+    def make_message_content_readable(self, message_content):
```

Comment:
The reason this feels strange to me is because removing `**` seems like a somewhat arbitrary rule, and it's not clear that when you have multiple words, turning `@**Vaibhav Singhal**` into `@Vaibhav Singhal` is what you want.

However, since you have been trying this, I think you should be the one to make this decision, and if you think removing `**` is the best option then I am happy with that one.

However, I would ask that you include a comment about the following:
- Why you do this
- Why it's okay to do this even when the mention contains multiple words
- The fact that this modified message uses a different key (because if this were in `content`, then this could cause problems with the bot library)